dnsext-do53/DNS/Do53/Cache.hs

```diff
@@ -67,8 +68,10 @@ import DNS.Types.Internal (TYPE (..))
 
 {- CRSet
    -  Left  - NXDOMAIN or NODATA, hold zone-domain delegation from
-   -  Right - not empty RRSET                                 -}
-type CRSet = Either Domain [RData]
+   -  Right (_:_, Nothing) - not empty RRSET, not verified
+   -  Right (_:_, Just []) - not empty RRSET, verification failed
+   -  Right (_:_, Just (_:_)) - not empty RRSET, verification succeeded -}
+type CRSet = Either Domain ([RData], Maybe [RD_RRSIG])
```

Comment:
It seems to me it's better to defined a proper data type rather than using `Either` and `Maybe`.
For instance,

```
data CRSet = NotExist Domain
           | NotVerified [RData]
           | VerifyFailed [RData]
           | Valid [RData] [RD_RRSIG]
```

Reply:
updated using inductive definition

https://github.com/kazu-yamamoto/dnsext/compare/8caca13613e498a57e94b407681f5874ef7578b7~6..8caca13613e498a57e94b407681f5874ef7578b7

Reply:
I think it would be clearer to have a separate structure for positive and negative, so I'll change it again